File: stepik-union/src/main/java/org/stepik/plugin/projectView/idea/IdeaTreeStructureProvider.java
package org.stepik.plugin.projectView.idea;

import com.intellij.psi.PsiClass;
import org.jetbrains.annotations.NotNull;
import org.stepik.plugin.projectView.StepikTreeStructureProvider;

/**
 * @author meanmail
 */
public class IdeaTreeStructureProvider extends StepikTreeStructureProvider {
    @Override
    protected boolean shouldAdd(@NotNull Object object) {
        return object instanceof PsiClass;
    }
}

File: stepik-core/src/main/kotlin/org/stepik/core/projectView/idea/IdeaTreeStructureProvider.kt
package org.stepik.core.projectView.idea

import com.intellij.psi.PsiClass
import org.stepik.core.projectView.StepikTreeStructureProvider


class IdeaTreeStructureProvider : StepikTreeStructureProvider() {
    override fun shouldAdd(any: Any): Boolean {
        return any is PsiClass
    }
}
